### PR TITLE
fix(startup): posthog issue when enabled and no connectivity

### PR DESCRIPTION
**Pull request by vijay-jangir**
Dockerfile enhancement, posthog bugfix

posthog doesn't let application start if there is any connection issue, temporarily fixed it to atleast get the application in started mode.
Updated dockerfile for ui and api so that it can run seemlessly on openshift update dockerfile for api for caching layers more efficiently

Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Comment by talboren** [comment]
@vijay-jangir I've edited the `EventCaptureMiddleware` to disable posthog completely unless otherwise configured since we're not really using it and the way you fixed it would've still caused the API to behave strange.

File: docker/Dockerfile.api [hunk 1]
```diff
@@ -17,11 +16,17 @@ ENV PIP_DEFAULT_TIMEOUT=100 \
...
+COPY keep keep
+COPY README.md README.md
 RUN poetry build && /venv/bin/pip install --use-deprecated=legacy-resolver dist/*.whl
 
 FROM base as final
 ENV PATH="/venv/bin:${PATH}"
 ENV VIRTUAL_ENV="/venv"
 COPY --from=builder /venv /venv
+# as per Openshift guidelines, https://docs.openshift.com/container-platform/4.11/openshift_images/create-images.html#use-uid_create-images
+RUN chgrp -R 0 /app && chmod -R g=u /app
```
Comment (talboren) [inline docker/Dockerfile.api:30]:
```suggestion
RUN chgrp -R 0 /app && chmod -R g=u /app
RUN chown -R keep:keep /app
```

**Comment by talboren** [comment]
@vijay-jangir this now lgtm, lmk once you've had a chance to take a look and I can merge this in.

**Review by talboren — APPROVED** [review]
LGTM

**Comment by vijay-jangir** [comment]
This looks fine, as long as we're okay with posthog still crashing the backend if network is behind a proxy or firewall, incase the POSTHOG_ENABLE is set to true.

**Comment by talboren** [comment]
> This looks fine, as long as we're okay with posthog still crashing the backend if network is behind a proxy or firewall, incase the POSTHOG_ENABLE is set to true.

It's not disabled by default, so it won't :)

**Pull request merged**